aeneas/src/ir/SsaNormalizer.v3

```diff
@@ -346,6 +346,306 @@ class SsaRaNormalizer extends SsaRebuilder {
+						for (i < width) {
+							var read = curBlock.opArrayGetElem(an.newType, Int.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos));
+							var longViewOp = V3Op.newIntViewI(read.getType(), Long.TYPE);
+							read = curBlock.addApply(app.source, longViewOp, [read]);
+							var shifted = curBlock.addApply(app.source, Long.TYPE.opShl(),
+								[read, newGraph.intConst(8 * i)]);
+							result = curBlock.addApply(app.source, Long.TYPE.opOr(),
+								[shifted, result]);
+						}
+						var enumTagLen = newGraph.intConst(x.enumDecl.cases.length);
+						var inBound = curBlock.opIntULt(IntType.!(result.getType()),
+							Int.TYPE, result, enumTagLen);
+						var split = SsaBlockSplit.new(context, curBlock);
+						curBlock = split.addIf(inBound);
+						curBlock = split.addElse();
+						curBlock = split.finish();
+						result = split.addPhi(Long.TYPE, [result, newGraph.longConst(0)]);
+						var intViewOp = V3Op.newIntViewI(result.getType(), nrtype);
+						result = curBlock.addApply(app.source, intViewOp, [result]);
+						return map1(app, result);
+					}
+					x: FloatType => {
+						var width = FloatType.!(ortype).byteSize();
+						for (i < width) {
+							var read = curBlock.opArrayGetElem(an.newType, Int.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos));
+							var shifted = curBlock.addApply(app.source, Long.TYPE.opShl(),
+								[read, newGraph.intConst(8 * i)]);
+							result = curBlock.addApply(app.source, Long.TYPE.opOr(),
+								[shifted, result]);
+						}
+						var floatViewOp: Operator;
+						if (FloatType.!(ortype).is64) {
+							floatViewOp = V3Op.newFloat64ViewI(result.getType());
+						}
+						else {
+							var intViewOp = V3Op.newIntViewI(result.getType(), Int.TYPE);
+							result = curBlock.addApply(app.source, intViewOp, [result]);
+							floatViewOp = V3Op.newFloat32ViewI(result.getType());
+						}
+						result = curBlock.addApply(app.source, floatViewOp, [result]);
+						return map1(app, result);
+					}
+					_ => ;
+				}
+			}
+			RefStructSetField(offset) => {
+				var newArgs = genRefs(args);
+				var array = newArgs[0], pos = newArgs[1];
+				match(newArgs[2].getType()) {
+					x: IntType => {
+						var width = IntType.!(x).byteSize();
+						//Terminal.put1("%d\n", width);
+						var results = Array<SsaInstr>.new(width);
+						def mask = newGraph.intConst(255);
+						for (i < width) {
+							var shifted = curBlock.addApply(app.source, IntType.!(x).opShr(),
+								[newArgs[2], newGraph.intConst(8 * i)]);
+							var value = curBlock.opIntAnd(shifted, mask);
+							value = curBlock.opIntViewI0(value.getType(), Byte.TYPE, value);
+							var write = curBlock.opArraySetElem(array.getType(), Byte.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos), value);
+							results[i] = write;
+						}
+						return mapN(app, results);
+					}
+					/* x: EnumType => {
```

Comment:
This code is still commented out. It should be very similar to the `int` case, except you might need to do a conversion from the enum type to an integer type first.